### ISSUE TITLE
Links and emphasis in the excerpt.

### ISSUE THREAD

**Issue by awulkan** [yarnjs/yarn-excerpt#1]
Would it be a good idea to maybe filter out links and emphasis tags from the excerpts? I think it can ruin the looks and user experience of a website to leave them in.

For example if there's a link right under the post title in an excerpt, or at the end before the "read more" link, it might cause people to click on that instead of the correct link which would take them to the wrong place.

Having bold and italicized text in the excerpt also prevents the page from having a clean look where all excerpts look uniform to each other. Same goes for having blue (or whatever link color you use) in the excerpts. It just doesn't look clean to me.

From an SEO standpoint it would also limit the amount of outgoing link juice from the index page by erasing the links.

Implementing this shouldn't be too difficult, right? I took a quick glance at the cheerio docs and it looks like you could just replace the `$.html(p).trim();` with `$.text(p).trim();`. I could be dead wrong on this though, so please correct me if I am. :)

These are just my two cents though. It would be nice to hear what you think about it.

**Comment by hswolff** [comment]
Personally I like having the text rich, but I can definitely understand your argument and desire to have just plain text.

I have [some undocumented code](http://github.com/reptar/reptar/blob/d7e63502592f04481320a08c25e4b90ad5ab4447/lib/plugin/index.js#L122) that allows you to pass additional configuration data to a plugin, so we could leverage that to allow for a user to decide which trim mode they want to use.

This would require a change to someone's own `_config.yml` file to:

``` yaml
plugins:
  excerpt:
    enabled: true
    options:
      textOnly: true
```

And the options object is passed to this plugin, and if that value `textOnly` is set to true then change what the plugin does. 

I'm completely down to allow for that flexibility.

**Comment by awulkan** [comment]
Looks like a good compromise. I'm still not up to speed on how Reptar really works, but I'll try to find some time to learn about plugins and their configs.  

Maybe you've already fixed it by then though. :)

**Comment by hswolff** [comment]
Haha, I think it's a good beginner fix. Would love to see some collaborators other than me! Take all the time you need to code on this!

**Comment by awulkan** [comment]
Pull request sent! #2

**Comment by hswolff** [comment]
Closed via #2
